Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -418,7 +418,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLRespon
     if (self.statusCode == 401
         && client.accessToken.refreshToken != nil
         && authenticateHeader
-        && [authenticateHeader rangeOfString:@"expired_token"].location != NSNotFound) {
+        && ([authenticateHeader rangeOfString:@"invalid_token"].location != NSNotFound || [authenticateHeader rangeOfString:@"invalid_request"].location != NSNotFound)) {
```

Comment:
Here's the authentication header in response to a 401, the debugger is paused at line 422:
Printing description of authenticateHeader:
Bearer realm="BetterUp", error="invalid_token", error_description="The access token is invalid"